### PR TITLE
Adding sponsorship doc and /sponsor redirect

### PR DESCRIPTION
**Pull request by JTWWilson**

### CODE REVIEW

File: _data/redirects.yaml [hunk 1]
```diff
@@ -2,6 +2,11 @@
 - source: /join
   destination: https://www.guildofstudents.com/studentgroups/societies/css/
 
+# join
```
Comment (AlexJBanks) [inline _data/redirects.yaml:5]: don't get why this is #join but its not the ned of the world

Comment (JTWWilson) [reply _data/redirects.yaml]: That was my fault for copy and pasting liberally - been corrected

**Comment by AlexJBanks** [comment]
Not sure that "Correct" is imperitive enough 🤔

**Comment by AlexJBanks** [comment]
redo the entire repo

**Review by AlexJBanks — APPROVED** [review]
jk its all good

**Pull request merged**